docs/ref/settings.txt

```diff
@@ -2440,6 +2440,10 @@ bandwidth but slows down performance. This is used by the ``CommonMiddleware``
 (see :doc:`/topics/http/middleware`) and in the``Cache Framework``
 (see :doc:`/topics/cache`).
 
+Using this setting is deprecated. If you want to use ETag-headers, you should use
```

Comment:
It's misleading to mark the setting as deprecated since it's required to opt-in to the behavior of `ConditionalGetMiddleware`. You should describe how the logic works and indicate that the setting won't have any effect after Django 2.0 since the E-tag behavior of `ConditionalGetMiddleware` will become mandatory.